src/Pool.sol

```diff
@@ -574,7 +586,7 @@ contract Erc20Pool is Pool {
 contract DaiPool is Erc20Pool {
     // solhint-disable no-empty-blocks
     /// @notice See `Erc20Pool` constructor documentation for more details.
-    constructor(uint64 cycleSecs, Dai dai) Erc20Pool(cycleSecs, dai) {}
+    constructor(uint64 cycleSecs, address dai) Erc20Pool(cycleSecs, dai) {}
```

Comment:
Ah maybe here I should have kept it to `IDai`.  Because the interface check for the `permit method would happen the first time in the updateSenderAndPermit method.

Reply:
Definitely it should be IDai. Then the constructor for Erc20Pool can be reverted to accepting IERC20s.